components/grade-table.js

```diff
@@ -1,32 +1,77 @@
 class GradeTable {
-  constructor(tableElement) {
+  constructor(tableElement, noGradesElement) {
     this.tableElement = tableElement;
+    this.noGradesElement = noGradesElement;
   }
   updateGrades(grades) {
-    var tbodyElement = this.tableElement.querySelector("tbody");
+    if (grades.length === 0) {
+      var pElement = this.noGradesElement;
+      pElement.classList.remove("d-none");
 
-    function removeAllChildNodes(parent) {
-      while (parent.firstChild) {
-        parent.removeChild(parent.firstChild);
-      }
-    };
+      var tbodyElement = this.tableElement.querySelector("tbody");
 
-    removeAllChildNodes(tbodyElement);
+      function removeAllChildNodes(parent) {
+        while (parent.firstChild) {
+          parent.removeChild(parent.firstChild);
+        }
+      };
 
-    for (var i = 0; i < grades.length; i++) {
-      var tableRow = document.createElement("tr");
+      removeAllChildNodes(tbodyElement);
 
-      var studentName = document.createElement("td");
-      studentName.textContent = grades[i].name;
+    } else {
 
-      var studentCourse = document.createElement("td");
-      studentCourse.textContent = grades[i].course;
+      if (grades.length !== 0) {
```

Comment:
Since the `else` only happens when `grades.length !== 0`, you don't need another `if (grades.length !== 0)` inside it.